subsys/net/ip/Kconfig

```diff
@@ -447,6 +447,39 @@ config NET_TCP_RECV_QUEUE_TIMEOUT
 	  SEQ 2. But if we receive SEQs 5,4,3,7 then the SEQ 7 is discarded
 	  because the list would not be sequential as number 6 is be missing.
 
+config NET_TCP_KEEPALIVE
+	bool "TCP Keep-Alives msg"
+	depends on NET_TCP
+	default y
+	help
+	  RFC1122#TCP Keep-Alives
+	  Enable TCP_KEEPIDLE, TCP_KEEPINTVL and TCP_KEEPCNT
+	  options processing. Note that TCP_KEEPIDLE and TCP_KEEPINTVL have to be set
+	  in seconds.
+
+config NET_TCP_KEEPIDLE_DEFAULT
+	int "TCP_KEEPIDLE value"
+	depends on NET_TCP_KEEPALIVE
+	default 5
+	help
+	  value in seconds.
+	  RFC1122#TCP Keep-Alives msg
```

Comment:
IMO it'd be good to elaborate a bit more in the help section about these options. The RFC1122 does not mention particualr keep-alive parameters, so it could be a bit misleading to reference it. Linux man pages have some good description for the options:

>TCP_KEEPCNT (since Linux 2.4) The maximum number of keepalive probes TCP should send before dropping the connection. This option should not be used in code intended to be portable.

>TCP_KEEPIDLE (since Linux 2.4) The time (in seconds) the connection needs to remain idle before TCP starts sending keepalive probes, if the socket option SO_KEEPALIVE has been set on this socket. This option should not be used in code intended to be portable.

>TCP_KEEPINTVL (since Linux 2.4) The time (in seconds) between individual keepalive probes. This option should not be used in code intended to be portable.